### PR TITLE
add support for opengraph tags

### PR DESCRIPTION
**Pull request by bakkot**
Fixes https://github.com/tc39/ecmarkup/issues/593:

Screenshot 2024-06-06 at 10 11 19 PM

I've hardcoded [the URL of the ECMA logo in this project's docs](https://tc39.es/ecmarkup/ecma-header.png) as the `og:image`, which isn't my favorite option, but it does work.

### CODE REVIEW

File: src/Spec.ts [hunk 1]
```diff
@@ -1438,6 +1438,41 @@ ${this.opts.multipage ? `<li><span>Navigate to/from multipage</span><code>m</cod
...
+    const insertMetaTag =
+      metas.length === 0
+        ? (node: HTMLMetaElement) => this.doc.head.prepend(node)
+        : (node: HTMLMetaElement) => metas[metas.length - 1].after(node);
+    if (!metas.some(n => n.getAttribute('property') === 'og:description')) {
+      const description =
+        this.opts.description ??
+        (this.doc.querySelector('emu-intro') ?? this.doc.querySelector('emu-clause'))?.textContent
+          ?.slice(0, 300)
+          .trim();
```
Comment (michaelficarra) [inline src/Spec.ts:1453]:
```suggestion
        (this.opts.description ?? (this.doc.querySelector('emu-intro') ?? this.doc.querySelector('emu-clause'))?.textContent)
          ?.slice(0, 300)
          .trim();
```

Maybe also slice the explicit description?

File: src/Spec.ts [hunk 2]
```diff
@@ -1438,6 +1438,41 @@ ${this.opts.multipage ? `<li><span>Navigate to/from multipage</span><code>m</cod
...
+        const meta = this.doc.createElement('meta');
+        meta.setAttribute('property', 'og:title');
+        meta.setAttribute('content', title);
+        insertMetaTag(meta);
+      }
+    }
+    if (!metas.some(n => n.getAttribute('property') === 'og:image')) {
+      const meta = this.doc.createElement('meta');
+      meta.setAttribute('property', 'og:image');
+      meta.setAttribute('content', 'https://tc39.es/ecmarkup/ecma-header.png');
```
Comment (michaelficarra) [inline src/Spec.ts:1473]: I don't love this choice, but it's better than nothing and we can improve it later.

What about the Ecma logo as a slightly less-bad option?

image

**Review by michaelficarra — APPROVED** [review]
LGTM otherwise

File: src/Spec.ts (hunk 2 above)
Comment (bakkot) [reply src/Spec.ts:1473]: Happy with that too though it would need to live somewhere static in order to be referenced. I can add it to the docs here.

Comment (bakkot) [reply src/Spec.ts]: Done. Further improvements of course welcome.

**Pull request merged**